File: src/main/java/dev/sky_lock/mocar/car/Capacity.java
package dev.sky_lock.mocar.car;

import java.util.Arrays;

/**
 * @author sky_lock
 */

public enum Capacity {
    ONE_SEAT(1),
    TWO_SEATS(2),
    FOR_SEATS(4);

    private final int value;

    Capacity(int value) {
        this.value = value;
    }

    static Capacity from(int value) {
        return Arrays.stream(values()).filter(capacity -> capacity.value() == value).findFirst().orElse(null);
    }

    public int value() {
        return value;
    }
}

File: src/main/kotlin/dev/sky_lock/pocketlifevehicle/vehicle/model/Capacity.kt
package dev.sky_lock.pocketlifevehicle.vehicle.model


/**
 * @author sky_lock
 */
enum class Capacity(private val value: Int) {
    ONE_SEAT(1), TWO_SEATS(2), FOUR_SEATS(4);

    fun value(): Int {
        return value
    }

    companion object {
        fun valueOf(value: Int): Capacity {
            return values().find { capacity: Capacity -> capacity.value() == value } ?: ONE_SEAT
        }
    }

}